tgui/packages/tgui/interfaces/ListInput.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * @file
+ * @copyright 2020 watermelon914 (https://github.com/watermelon914)
+ * @license MIT
+ */
+
+import { clamp01 } from 'common/math';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Flex, Section, Input } from '../components';
+import { Window } from '../layouts';
+
+let lastScrollTime = 0;
+
+export const ListInput = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    title,
+    message,
+    buttons,
+    timeout,
+  } = data;
+
+  // Search
+  const [showSearchBar, setShowSearchBar] = useLocalState(context, "search_bar", false);
+  const [displayedArray, setDisplayedArray] = useLocalState(context, "displayed_array", buttons);
+
+  // KeyPress
+  const [searchArray, setSearchArray] = useLocalState(context, "search_array", []);
+  const [searchIndex, setSearchIndex] = useLocalState(context, "search_index", 0);
+  const [lastCharCode, setLastCharCode] = useLocalState(context, "last_char_code", null);
+
+  // Selected Button
+  const [selectedButton, setSelectedButton] = useLocalState(context, "selected_button", null);
+  return (
+    <Window
+      title={title}
+      width={325}
+      height={325}
+    >
+      {timeout !== undefined && <Loader value={timeout} />}
+      <Window.Content>
+        <Section fill>
+          <Flex
```

Comment:
This flex can be put above the section. Rule of thumb - first lay out compartments with flex, then fill them with sections.